src/number-formatting/index.ts

```diff
@@ -5,6 +5,9 @@ import shortFormat from './short-format';
 
 const numberOfZerosRx = /0+$/;
 
+// browsers [ie,chromium,firefox] start rounding of decimal number after 16-17 chars.

```

Comment:
its not browser specific, I'm pretty sure it is javascript specific.

did you come up with this number through experimentation or some resource? Can you just expand on how this number was arrived at, so that if we need to adjust it we know how we got this?

Reply:
We have come up with this number through experimentation.
Simply in console if we have number greater than 16/17 chars  then js starts to round off
for ex- let c = `123456789123456789` , c becomes `123456789123456780`

`123456789123456.789` becomes `123456789123456.78`

`1234567891234567.1` becomes `1234567891234567`
`1234567891234567.8` becomes `1234567891234567.8`


+ if number is greater than 21 than it is converted to exponential notation

cc: @yugandhar02

Reply:
Ok - please add this to the comment and it’s good to go

Reply:
Done